src/detect-parse.c

```diff
@@ -433,7 +433,8 @@ SigMatch *DetectGetLastSMFromMpmLists(const DetectEngineCtx *de_ctx, const Signa
     uint32_t sm_type;
 
     /* if we have a sticky buffer, use that */
-    if (s->init_data->list != DETECT_SM_LIST_NOTSET) {
+    if (s->init_data->list != DETECT_SM_LIST_NOTSET &&
+            s->init_data->list < (int)s->init_data->smlists_array_size) {
```

Comment:
I wonder if this masks a deeper bug? Should we have rejected a rule earlier or perhaps updated `smlists_array_size` earlier?

Reply:
I just copied your idea from commit https://github.com/OISF/suricata/commit/e902aaf838acb552350078070092a2c61b918b19

Should `alert sip any any -> any any (frame:request.headers;fast_pattern; content: "toto";)` be a valid rule ?